lttngpy/src/lttngpy/snapshot.cpp

```
@@ -0,0 +1,37 @@
+#include <lttng/snapshot.h>
+#include <lttng/lttng-error.h>
+
+#include <string>
+
+#include "lttngpy/snapshot.hpp"
+
+namespace lttngpy
+{
+
+int lttng_record_snapshot(
+  const std::string & session_name,
+  const uint32_t id,
+  const uint64_t max_size,
+  const std::string & name,
+  const std::string & url,
+  const int wait)
+{
+  struct lttng_snapshot_output * output = lttng_snapshot_output_create();
+  if (nullptr == output) {
+    return -LTTNG_ERR_UNK;
+  }
+
+  // Set snapshot output attributes.
+  lttng_snapshot_output_set_id(id, output);
+  lttng_snapshot_output_set_size(max_size, output);
+  lttng_snapshot_output_set_name(name.c_str(), output);  
+  lttng_snapshot_output_set_local_path(url.c_str(), output);
+
+  // Add snapshot output to the session.
+  lttng_snapshot_add_output(session_name.c_str(), output);
```

I know it's annoying to write and ugly to read, but you should check the return code of each of these calls and return right away if there is an error, i.e., `if (0 != ret)` then `return ret;`. There probably won't be an error here, but if there is an error with any of these, we generally want to report the exact error as soon as possible, which means returning the error/return code.

---

Added this in https://github.com/ros2/ros2_tracing/pull/191/commits/b76c5eda1ea4c26a6c9b530b6af0cddf4ba3ef06

---

I think we might get errors in `lttng_snapshot_add_output`, but we shouldn't get any errors in `lttng_snapshot_output_set_...` right (since these functions are just setting the values)?

---

we shouldn't get an error with `lttng_snapshot_output_set_*()`, but we could according to the API docs (f the inputs are invalid), so we should check